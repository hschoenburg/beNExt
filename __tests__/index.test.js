/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Link from 'next/link'

import Home from '../pages/index.js'

import Router from 'next/router'
const mockedRouter = { push: () => {}, prefetch: () => {} }
Router.router = mockedRouter

describe('With Enzyme', () => {
  it('Home shows 3 Cards each with a Link', () => {
    const wrap = shallow(<Home />)
    expect(wrap.find('a.card').length).toBe(3)
    expect(wrap.find(Link).length).toBe(3)
  })
})

describe('With Snapshot Testing', () => {
  it('Home shows 3 cards', () => {
    const component = renderer.create(<Home />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
