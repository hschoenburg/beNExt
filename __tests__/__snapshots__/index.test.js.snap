// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Home shows 3 cards 1`] = `
<div
  className="jsx-243749409"
>
  <nav
    className="jsx-2055897931"
  >
    <ul
      className="jsx-2055897931"
    >
      <li
        className="jsx-2055897931"
      >
        <a
          className="jsx-2055897931"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <ul
        className="jsx-2055897931"
      >
        <li
          className="jsx-2055897931"
        >
          <a
            className="jsx-2055897931"
            href="/explorer"
            onClick={[Function]}
          >
            Explorer
          </a>
        </li>
        <li
          className="jsx-2055897931"
        >
          <a
            className="jsx-2055897931"
            href="/market"
            onClick={[Function]}
          >
            Market
          </a>
        </li>
      </ul>
    </ul>
  </nav>
  <div
    className="jsx-243749409 hero"
  >
    <h1
      className="jsx-243749409 title"
    >
      Welcome to Block Explorer
    </h1>
    <p
      className="jsx-243749409 description"
    >
      Building the next great crypto news and tools website.
    </p>
    <div
      className="jsx-243749409 row"
    >
      <a
        className="jsx-243749409 card"
        href="/explorer"
        onClick={[Function]}
      >
        <h3
          className="jsx-243749409"
        >
          Explorer
        </h3>
        <p
          className="jsx-243749409"
        >
          Bitcoin transactions, addresses and blocks oh my. Backend - bcoin
        </p>
      </a>
      <a
        className="jsx-243749409 card"
        href="/market"
        onClick={[Function]}
      >
        <h3
          className="jsx-243749409"
        >
          Market
        </h3>
        <p
          className="jsx-243749409"
        >
          XRP, ETH, BTC prices prices prices. Backend - CoinMarketCap and more
        </p>
      </a>
      <a
        className="jsx-243749409 card"
        href="/news"
        onClick={[Function]}
      >
        <h3
          className="jsx-243749409"
        >
          News
        </h3>
        <p
          className="jsx-243749409"
        >
          Headlines, articles, news. Backend - Wordpress 
        </p>
      </a>
    </div>
  </div>
</div>
`;
